### PR TITLE
Add support for customizing terminationGracePeriodSeconds of the VTTablet pods

### PR DESCRIPTION
**Pull request by yoheimuta**
This PR introduces support for customizing the terminationGracePeriodSeconds of the VTTablet pods. Currently, this value is set to 30 minutes.

Having [a fairly high terminationGracePeriodSeconds](https://github.com/planetscale/vitess-operator/blob/9cb55b422e21e6fbea1c3614c4bfdedd3a2f963f/pkg/operator/vttablet/constants.go#L51) can lead to issues when adding sidecars that don't handle SIGTERM properly, resulting in long delays during the termination process. At least, this default constant value is not suitable for the use case of unmanaged tablets.

Similar to https://github.com/planetscale/vitess-operator/pull/406.

### CODE REVIEW

File: pkg/operator/vttablet/pod.go [hunk 1]
```diff
@@ -313,7 +313,11 @@ func UpdatePod(obj *corev1.Pod, spec *Spec) {
 		obj.Spec.SecurityContext.FSGroup = pointer.Int64Ptr(planetscalev2.DefaultVitessFSGroup)
 	}
 
-	obj.Spec.TerminationGracePeriodSeconds = pointer.Int64Ptr(terminationGracePeriodSeconds)
+	if spec.Vttablet.TerminationGracePeriodSeconds != nil {
+		obj.Spec.TerminationGracePeriodSeconds = spec.Vttablet.TerminationGracePeriodSeconds
+	} else {
+		obj.Spec.TerminationGracePeriodSeconds = pointer.Int64Ptr(terminationGracePeriodSeconds)
```
Comment (mattlord) [inline pkg/operator/vttablet/pod.go:319]: Nit, but I feel like we should rename this variable to `defaultTerminationGracePeriodSeconds` now.

**Review by mattlord — APPROVED** [review]
Thanks, @yoheimuta ! I just had one nit about the existing var name. Let me know what you think.

**Review by frouioui — APPROVED** [review]
Looks good to me, thank you @yoheimuta

**Comment by frouioui** [comment]
Closing and re-opening to re-trigger the CI tests that are in `Expected`

**Pull request merged**